drivers/flash/flash_stm32wbx.c

```diff
@@ -97,17 +197,100 @@ static int erase_page(struct device *dev, uint32_t page)
 		return rc;
 	}
 
-	/* Check erase operation allowed */
-	if (regs->CR & FLASH_SR_PESD) {
-		return -EBUSY;
-	}
+	/* Implementation of STM32 AN5289, proposed in STM32WB Cube Application
+	 * BLE_RfWithFlash
+	 * https://github.com/STMicroelectronics/STM32CubeWB/tree/master/Projects/P-NUCLEO-WB55.Nucleo/Applications/BLE/BLE_RfWithFlash
+	 */
+
+	do {
+		/**
+		 * When the PESD bit mechanism is used by CPU2 to protect its
+		 * timing, the PESD bit should be polled here.
+		 * If the PESD is set, the CPU1 will be stalled when reading
+		 * literals from an ISR that may occur after the flash
+		 * processing has been requested but suspended due to the PESD
+		 * bit.
+		 *
+		 * Note: This code is required only when the PESD mechanism is
+		 * used to protect the CPU2 timing.
+		 * However, keeping that code make it compatible with both
+		 * mechanisms.
+		 */
```

Comment:
Not a blocker, but I think that mechanism details (i.e. all detailed comments) could go in a single place as a sort of documentation. In my opinion there's too much comments inline with the code.

Reply:
As code and comments are inherited from https://github.com/STMicroelectronics/STM32CubeWB/blob/master/Projects/P-NUCLEO-WB55.Nucleo/Applications/BLE/BLE_RfWithFlash/Core/Src/flash_driver.c
I would prefer to keep it as is. Especially for further maintenance.